src/software/geom/algorithms/contains.cpp

```diff
@@ -61,13 +62,13 @@ bool contains(const Polygon& container, const Point& contained)
 bool contains(const Ray& container, const Point& contained)
 {
     Point point_in_ray_direction = container.getStart() + container.toUnitVector();
-
-    bool point_is_ray_start = contained == container.getStart();
+    bool point_is_ray_start      = contained == container.getStart();
     bool point_collinear_with_ray =
-        collinear(contained, container.getStart(), point_in_ray_direction);
-    bool point_is_in_ray_direction =
+        collinear(contained, container.getStart(), point_in_ray_direction, 1e-9, 200);
+    bool point_is_in_ray_direction = almostEqual(
         ((contained - container.getStart()).normalize() - container.toUnitVector())
-            .length() < FIXED_EPSILON;
+            .length(),
+        0, 1e-9, 200);
```

Comment:
is it possible to use geom constants or some multiple of one, e.g. `2*FIXED_EPSILON`, so we don't have magic numbers in our code?

Reply:
used `1000 * FIXED_EPSILON` and `2 * ULPS_EPSILON_HUNDRED`, which work fine

Reply:
used `1000 * FIXED_EPSILON` and `2 * ULPS_EPSILON_HUNDRED`, which work fine